nautobot/dcim/factory.py

```diff
@@ -85,49 +156,197 @@ class Meta:
+    def content_types(self, create, extract, **kwargs):
+        """Assign some contenttypes to a location after generation"""
+        if self.name in ["Root", "Campus"]:
+            self.content_types.set(ContentType.objects.filter(FeatureQuery("locations").get_query()))
+        elif self.name in ["Building", "Floor"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Prefix),
+                    ContentType.objects.get_for_model(Rack),
+                    ContentType.objects.get_for_model(RackGroup),
+                    ContentType.objects.get_for_model(VLANGroup),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Room"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Cluster),
+                    ContentType.objects.get_for_model(PowerPanel),
+                    ContentType.objects.get_for_model(Rack),
+                    ContentType.objects.get_for_model(RackGroup),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Elevator"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Cluster),
+                    ContentType.objects.get_for_model(PowerPanel),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Aisle"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(CircuitTermination),
+                    ContentType.objects.get_for_model(Device),
+                    ContentType.objects.get_for_model(PowerPanel),
+                    ContentType.objects.get_for_model(Rack),
+                    ContentType.objects.get_for_model(RackGroup),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+
+
+class LocationFactory(DjangoModelFactory):
+    class Meta:
+        model = Location
+        exclude = (
+            "has_parent",
+            "has_site",
+            "has_tenant",
+            "has_description",
+            "_parent",
+        )
+
+    name = factory.LazyAttributeSequence(lambda l, n: f"{l.location_type.name}-{n:02d}")
+    status = random_instance(lambda: Status.objects.get_for_model(Location), allow_null=False)
+
+    @factory.iterator
+    def location_type():  # pylint: disable=no-method-argument
+        lts = LocationType.objects.all()
+        for lt in lts:
+            yield lt
+
+    @factory.lazy_attribute
+    def parent(self):
+        """
+        The parent attribute of all the location types other than root and campus are deterministic.
+        There is a 50% chance whether a root or campus type location (both nestable) have a parent.
+        """
+        candidate_parents = Q()
```

Comment:
`Q()` matches all objects. Instead this should probably be something like `Q(pk=None)`